streaming/streaming.go

```diff
@@ -304,22 +304,54 @@ func (c *Conn) receivedOrderBook(ob orderBook) error {
 }
 
 func (c *Conn) receivedUpdate(u Update) error {
-	c.mu.Lock()
-	defer c.mu.Unlock()
+	valid, err := c.validateUpdate(u)
+	if err != nil {
+		return err
+	}
+
+	// If update is not valid, ignore
+	if !valid {
+		return nil
+	}
+
+	err = c.processUpdate(u)
+	if err != nil {
+		return err
+	}
+
+	if c.updateCallback != nil {
+		c.updateCallback(u)
+	}
+
+	return nil
+}
+
+func (c *Conn) validateUpdate(u Update) (bool, error) {
```

Comment:
I don't think we should unlock the mutex between validating and processing the update, can you move these two functions into one. then you can call the callback after.

Reply:
Made the change as requested.